### PR TITLE
integration test for cross cloud b2b

### PR DESCRIPTION
**Pull request by SomkaPe**

### CODE REVIEW

**Review by Avery-Dunn — APPROVED** [review]

File: src/integrationtest/java/com.microsoft.aad.msal4j/AcquireTokenInteractiveIT.java [hunk 1]
```diff
@@ -168,6 +169,77 @@ private void assertAcquireTokenInstanceAware(User user) {
...
+    // uncomment when mooncake account will be available
+    public void acquireTokensInHomeAndGuestClouds_MooncakeAccount() throws MalformedURLException, ExecutionException, InterruptedException {
+        String AUTHORITY_MOONCAKE = "https://login.chinacloudapi.cn/mncmsidlab1.partner.onmschina.cn";
+
+        acquireTokensInHomeAndGuestClouds(AzureEnvironment.AZURE_CHINA, AUTHORITY_MOONCAKE);
+    }
+
+    public void acquireTokensInHomeAndGuestClouds(String homeCloud, String homeCloudAuthority) throws MalformedURLException, ExecutionException, InterruptedException {
+        String AUTHORITY_PUBLIC_TENANT_SPECIFIC = "https://login.microsoftonline.com/msidlab4.onmicrosoft.com";
+        String SCOPE = "user.read";
```
Comment (sangonzal) [inline src/integrationtest/java/com.microsoft.aad.msal4j/AcquireTokenInteractiveIT.java:189]: You can reuse this from `TestConstants`

File: src/integrationtest/java/infrastructure/SeleniumExtensions.java [hunk 2]
```diff
@@ -81,8 +88,33 @@ public static void performADLogin(WebDriver driver, User user){
...
 
         LOG.info("Loggin in ... click submit");
-         waitForElementToBeVisibleAndEnable(driver, new By.ById(fields.getPasswordSigInButtonId())).
+        waitForElementToBeVisibleAndEnable(driver, new By.ById(fields.getPasswordSigInButtonId())).
                 click();
+
+        try {
+            checkAuthenticationCompletePage(driver);
+            return;
+        } catch (TimeoutException ex) {
```
Comment (sangonzal) [inline src/integrationtest/java/infrastructure/SeleniumExtensions.java:97]: If there is a timeout exception, should we erroring out with a helpful message? There is nothing in the catch block here or in line 105

File: src/integrationtest/java/labapi/LabService.java [hunk 3]
```diff
@@ -59,7 +59,13 @@ User getUser(UserQueryParameters query){
 
             User[] users = convertJsonToObject(result, User[].class);
             User user = users[0];
-            user.setPassword(getSecret(user.getLabName()));
+            if(user.getUserType().equals("Guest")){
```
Comment (sangonzal) [inline src/integrationtest/java/labapi/LabService.java:62]: "guest" is used in a couple of places, can be into a constant.

**Review by sangonzal — APPROVED** [review]

File: src/integrationtest/java/infrastructure/SeleniumExtensions.java (hunk 2 above)
Comment (SomkaPe) [reply src/integrationtest/java/infrastructure/SeleniumExtensions.java:97]: depending on types of account and authority authentication can be completed or question about "Staying logged in" might appears. Both option are valid.

**Pull request merged**